pageserver/src/waldecoder.rs

```diff
@@ -382,6 +383,184 @@ impl XlHeapUpdate {
     }
 }
 
+///
+/// Note: Parsing some fields is missing, because they're not needed.
+/// FIXME: abort rels in twophase record is missing and is needed
+///
+/// This mirrors the xl_xact_parsed_commit struct in PostgreSQL
+///
+#[derive(Debug)]
+pub struct XlXactParsedCommit {
```

Comment:
Why there is no XID in XlXactParsedCommit/Abort?
Pleas notice that almost the same code is used for XLOG_XACT_COMMIT_PREPARED/XLOG_XACT_ABORT_PREPARED.
Also code for abort/commit seems to be almost identical. Should we really use two difference structs and two difference parse  methods for them or it will be enough to reuse the same code for both?

Reply:
> Why there is no XID in XlXactParsedCommit/Abort?

The XID is in the XLogRecord header, it's not part of the payload. Except in case of COMMIT_PREPARED, then it's stored in the twophase_xid field, but I left that out for now. Yeah, that should be added too. I modeled this after PostgreSQL's `xl_xact_parsed_commit` struct.

> Please notice that almost the same code is used for XLOG_XACT_COMMIT_PREPARED/XLOG_XACT_ABORT_PREPARED.
> Also code for abort/commit seems to be almost identical. Should we really use two difference structs and two difference parse methods for them or it will be enough to reuse the same code for both?

Makes sense to use the same structs and code for XLOG_XACT_COMMIT and XLOG_XACT_COMMIT_PREPARED.